lib/meta-box-partial-page.php

```diff
@@ -0,0 +1,607 @@
+	 * @since wp-core 2.1.0
+	 */
+	// @codingStandardsIgnoreStart
+	do_action( "admin_head-{$hook_suffix}" );
+	// @codingStandardsIgnoreEnd
+
+	/**
+	 * Fires in head section for all admin pages.
+	 *
+	 * @since wp-core 2.1.0
+	 */
+	do_action( 'admin_head' );
+
+	/**
+	 * The main way post.php sets body class.
+	 */
+	if ( get_user_setting( 'mfold' ) == 'f' ) {
+		$admin_body_class .= ' folded';
+	}
+
+	if ( ! get_user_setting( 'unfold' ) ) {
+		$admin_body_class .= ' auto-fold';
+	}
+
+	if ( is_admin_bar_showing() ) {
+		$admin_body_class .= ' admin-bar';
+	}
+
+	if ( is_rtl() ) {
+		$admin_body_class .= ' rtl';
+	}
+
+	if ( $current_screen->post_type ) {
+		$admin_body_class .= ' post-type-' . $current_screen->post_type;
+	}
+
+	if ( $current_screen->taxonomy ) {
+		$admin_body_class .= ' taxonomy-' . $current_screen->taxonomy;
+	}
+
+	$admin_body_class .= ' branch-' . str_replace( array( '.', ',' ), '-', floatval( get_bloginfo( 'version' ) ) );
+	$admin_body_class .= ' version-' . str_replace( '.', '-', preg_replace( '/^([.0-9]+).*/', '$1', get_bloginfo( 'version' ) ) );
+	$admin_body_class .= ' admin-color-' . sanitize_html_class( get_user_option( 'admin_color' ), 'fresh' );
+	$admin_body_class .= ' locale-' . sanitize_html_class( strtolower( str_replace( '_', '-', get_user_locale() ) ) );
+
+	if ( wp_is_mobile() ) {
+		$admin_body_class .= ' mobile';
+	}
+
+	if ( is_multisite() ) {
+		$admin_body_class .= ' multisite';
+	}
+
+	if ( is_network_admin() ) {
+		$admin_body_class .= ' network-admin';
+	}
+
+	$admin_body_class .= ' no-customize-support no-svg';
+
+	?>
+	</head>
+	<?php
+
+	/**
+	 * Filters the CSS classes for the body tag in the admin.
+	 *
+	 * This filter differs from the {@see 'post_class'} and {@see 'body_class'} filters
+	 * in two important ways:
+	 *
+	 * 1. `$classes` is a space-separated string of class names instead of an array.
+	 * 2. Not all core admin classes are filterable, notably: wp-admin, wp-core-ui,
+	 *    and no-js cannot be removed.
+	 *
+	 * @since wp-core 2.3.0
+	 *
+	 * @param string $classes Space-separated list of CSS classes.
+	 */
+	$admin_body_classes = apply_filters( 'admin_body_class', '' );
+
+	// This page should always match up with the edit action.
+	$action = 'edit';
+
+	?>
+	<body class="wp-admin wp-core-ui no-js <?php echo $admin_body_classes . ' ' . $admin_body_class; ?>">
+	<script type="text/javascript">
+		document.body.className = document.body.className.replace('no-js','js');
+	</script>
+	<?php
+}
+
+/**
+ * This matches the portion of creating a form found in edit-form-advanced.php.
+ *
+ * Code starts roughly around line 500.
+ *
+ * @since 1.5.0
+ *
+ * @param WP_Post $post     Current post object.
+ * @param string  $location Metabox location: one of 'normal', 'advanced', 'side'.
+ */
+function gutenberg_meta_box_partial_page_post_form( $post, $location ) {
+	$notice     = false;
+	$form_extra = '';
+	if ( 'auto-draft' === $post->post_status ) {
+		if ( 'edit' === $action ) {
```

Comment:
`$action` is not assigned in this scope.

```
Notice: Undefined variable: action in /srv/www/editor/htdocs/wp-content/plugins/gutenberg/lib/meta-box-partial-page.php on line 345
```

Reply:
Woops.

Reply:
https://github.com/WordPress/gutenberg/pull/3092